### ISSUE TITLE
Deno Bundle breaks WASM ABI - Upgrade SWC

### ISSUE THREAD

**Issue by AaronO** [denoland/deno#9199]
## Problem

I've been building a `deno` app using TS & WASM built with `rust` & `wasm-bindgen` and I noticed that `deno bundle` and `deno compile` break the "ABI" specifically `addHeapObject()` thus breaking pointers passed between JS <> WASM.

`addHeapObject()` is corrupted when transpiled, specifically in latter half of the function, `idx` is incorrectly swapped for `heap_next`. This appears to be fixed in the latest version of `swc` (see context and code below).

## Solution

This should be solved by upgrading `swc`, it might make sense to add a bundle fixture to test for this given how central `wasm-bindgen`'s ABI is to most WASM modules.

## Context

```js
// ❯ cat heap.js
const heap = new Array(32).fill(undefined);
heap.push(undefined, null, true, false);
let heap_next = heap.length;
function addHeapObject(obj) {
    if (heap_next === heap.length) heap.push(heap.length + 1);
    const idx = heap_next;
    heap_next = heap[idx];
    heap[idx] = obj;
    return idx;
}
addHeapObject("hello");

// ❯ deno bundle heap.js
// Bundle file:///.../heap.js
const heap = new Array(32).fill(undefined);
heap.push(undefined, null, true, false);
let heap_next = heap.length;
function addHeapObject(obj) {
    if (heap_next === heap.length) heap.push(heap.length + 1);
    const idx = heap_next;
    heap_next = heap[heap_next];
    heap[heap_next] = obj;
    return heap_next;
}
addHeapObject("hello");

// ❯ npx swc heap.js
var heap = new Array(32).fill(undefined);
heap.push(undefined, null, true, false);
var heap_next = heap.length;
function addHeapObject(obj) {
    if (heap_next === heap.length) heap.push(heap.length + 1);
    var idx = heap_next;
    heap_next = heap[idx];
    heap[idx] = obj;
    return idx;
}
addHeapObject("hello");
```

**Comment by AaronO** [comment]
This bug actually isn't fixed by upgrading SWC (just tried it locally),

## Code to reproduce

```ts
const HEAP_SRC = `
const heap = new Array(32).fill(undefined);
heap.push(undefined, null, true, false);
let heap_next = heap.length;
function addHeapObject(obj) {
    if (heap_next === heap.length) heap.push(heap.length + 1);
    const idx = heap_next;
    heap_next = heap[idx];
    heap[idx] = obj;
    return idx;
}
addHeapObject("hello");
`;

async function main() {
    const transpiled = await Deno.emit('file:///heap.js', {
        sources: { 'file:///heap.js': HEAP_SRC },
    });
    console.log(transpiled.files['file:///heap.js.js']);
    
    console.log();
    console.log('---');
    console.log();
    
    const bundled = await Deno.emit('file:///heap.js', {
        bundle: 'esm',
        sources: { 'file:///heap.js': HEAP_SRC },
    });
    console.log(bundled.files['deno:///bundle.js']);
}
main();
```

## Output

```
~/git/deno/target/release/deno run --unstable -A ./t.ts
```

Outputs:

```js
"use strict";
const heap = new Array(32).fill(undefined);
heap.push(undefined, null, true, false);
let heap_next = heap.length;
function addHeapObject(obj) {
    if (heap_next === heap.length)
        heap.push(heap.length + 1);
    const idx = heap_next;
    heap_next = heap[idx];
    heap[idx] = obj;
    return idx;
}
addHeapObject("hello");

---

const heap = new Array(32).fill(undefined);
heap.push(undefined, null, true, false);
let heap_next = heap.length;
function addHeapObject(obj) {
    if (heap_next === heap.length) heap.push(heap.length + 1);
    const idx = heap_next;
    heap_next = heap[idx];
    heap[idx] = obj;
    return heap_next;
}
addHeapObject("hello");
```

**Comment by kitsonk** [comment]
cc/ @kdy1

**Comment by bartlomieju** [comment]
Should be fixed in https://github.com/denoland/deno/pull/9374

**Comment by AaronO** [comment]
@bartlomieju I can confirm this is now fixed.

However I believe it was fixed in `1.7.1` (via #9251)
